app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save {self.email = email.downcase}
+  validates :name, presence: true, length: {maximun: Settings.validate.name.maximum}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: Settings.validate.email.maximum},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
I have another question: What is has_secure_password and why you used it here?

Reply:
It can validate password automatically. I used it here to validate password for User

Reply:
@maiq721 I am not satisfied about the answer. Could you please study more and add more words to your answer.

Reply:
I got it correctlly... has_secure_password is the method encrypt password with Bcrypt, add a password attribute(read-only) to the model,ensure that password_digest is present before model is saved, validates that the password is confirmed(you have to confirm password when you register an account on some web site) and authenticate password. I thought all of them are validate.